src/test/app/Batch_test.cpp

```diff
@@ -0,0 +1,3047 @@
+                    return jv;
+                };
+
+            auto const batchFee = batch::calcBatchFee(env, 0, 2);
+            auto const seq = env.seq(alice);
+            env(batch::outer(alice, seq, batchFee, tfAllOrNothing),
+                batch::inner(ammCreate(XRP(10), USD(10)), seq + 1),
+                batch::inner(pay(alice, bob, XRP(10)), seq + 2),
+                ter(telINSUF_FEE_P));
+            env.close();
+        }
+    }
+
+    void
+    testCalculateBaseFee(FeatureBitset features)
+    {
+        testcase("calculate base fee");
+
+        using namespace test::jtx;
+        using namespace std::literals;
+
+        // telENV_RPC_FAILED: Batch: txns array exceeds 8 entries.
+        {
+            test::jtx::Env env{*this, envconfig()};
+
+            auto const alice = Account("alice");
+            auto const bob = Account("bob");
+            env.fund(XRP(1000), alice, bob);
+            env.close();
+
+            auto const batchFee = batch::calcBatchFee(env, 0, 9);
+            auto const aliceSeq = env.seq(alice);
+            env(batch::outer(alice, aliceSeq, batchFee, tfAllOrNothing),
+                batch::inner(pay(alice, bob, XRP(1)), aliceSeq),
+                batch::inner(pay(alice, bob, XRP(1)), aliceSeq),
+                batch::inner(pay(alice, bob, XRP(1)), aliceSeq),
+                batch::inner(pay(alice, bob, XRP(1)), aliceSeq),
+                batch::inner(pay(alice, bob, XRP(1)), aliceSeq),
+                batch::inner(pay(alice, bob, XRP(1)), aliceSeq),
+                batch::inner(pay(alice, bob, XRP(1)), aliceSeq),
+                batch::inner(pay(alice, bob, XRP(1)), aliceSeq),
+                batch::inner(pay(alice, bob, XRP(1)), aliceSeq),
+                ter(telENV_RPC_FAILED));
+            env.close();
+        }
+
+        // temARRAY_TOO_LARGE: Batch: txns array exceeds 8 entries.
+        {
+            test::jtx::Env env{*this, envconfig()};
+
+            auto const alice = Account("alice");
+            auto const bob = Account("bob");
+            env.fund(XRP(1000), alice, bob);
+            env.close();
+
+            auto const batchFee = batch::calcBatchFee(env, 0, 9);
+            auto const aliceSeq = env.seq(alice);
+            auto jt = env.jtnofill(
+                batch::outer(alice, aliceSeq, batchFee, tfAllOrNothing),
+                batch::inner(pay(alice, bob, XRP(1)), aliceSeq),
+                batch::inner(pay(alice, bob, XRP(1)), aliceSeq),
+                batch::inner(pay(alice, bob, XRP(1)), aliceSeq),
+                batch::inner(pay(alice, bob, XRP(1)), aliceSeq),
+                batch::inner(pay(alice, bob, XRP(1)), aliceSeq),
+                batch::inner(pay(alice, bob, XRP(1)), aliceSeq),
+                batch::inner(pay(alice, bob, XRP(1)), aliceSeq),
+                batch::inner(pay(alice, bob, XRP(1)), aliceSeq),
+                batch::inner(pay(alice, bob, XRP(1)), aliceSeq));
+
+            env.app().openLedger().modify(
```

Comment:
Why can't this test be submitted normally?

Reply:
They can and I believe they are just above. This is just testing the injection of a transaction which bypasses the rpc.